mapviz_plugins/src/pointcloud2_plugin.cpp

```diff
@@ -271,13 +281,16 @@ namespace mapviz_plugins
 
   void PointCloud2Plugin::UpdateColors()
   {
-    std::deque<Scan>::iterator scan_it = scans_.begin();
-    for (; scan_it != scans_.end(); ++scan_it)
     {
-      std::vector<StampedPoint>::iterator point_it = scan_it->points.begin();
-      for (; point_it != scan_it->points.end(); point_it++)
+      QMutexLocker locker(&scan_mutex_);
+      std::deque<Scan>::iterator scan_it = scans_.begin();
+      for (; scan_it != scans_.end(); ++scan_it)
       {
-        point_it->color = CalculateColor(*point_it);
+        std::deque<StampedPoint>::iterator point_it = scan_it->points.begin();
```

Comment:
I believe @nick-alton wrote this originally, so probably a question for him.  I'd guess it's just to make the `for` line shorter.